### PR TITLE
[doc] update DV and DOC wording from sanity to smoke

### PR DESCRIPTION
**Pull request by cindychip**
1. For all the IPs and top-level dv, this commit update the word from sanity to smoke.
For OTBN, I only changed under the `dv/uvm` folder. @rswarbrick please let me know if
 I break anything.

2. Update all gen-script to use smoke.

3. Update RTL checklist from SANITY to CROSS

I think in order to pass Private CI check, I will need to update the private CI command.

Signed-off-by: Cindy Chen <chencindy@google.com>

### CODE REVIEW

File: hw/dv/doc/dv_plan_template.md [hunk 1]
```diff
@@ -109,10 +109,10 @@ It creates the following analysis ports to retrieve the data monitored by corres
 ## Building and running tests
 We are using our in-house developed [regression tool]({{< relref "hw/dv/tools/README.md" >}}) for building and running our tests and regressions.
 Please take a look at the link for detailed information on the usage, capabilities, features and known issues.
-Here's how to run a basic sanity test:
+Here's how to run a basic smoke test:
```
Comment (rswarbrick) [inline hw/dv/doc/dv_plan_template.md:112]: Nit: maybe this should just be "... run a smoke test"? I'm not sure what "basic" contributes here any more.

**Review by rswarbrick — COMMENTED** [review]
Hi Cindy,

Thanks for doing this. I'm not entirely sure about the find-and-replace across the tree approach: there's at least one change in a comment in the OTBN code that didn't really make sense. I wonder whether we should instead grep in each IP block and pester the authors until they fix it themselves?

Since the OTBN stuff is in a bit of a state of flux, I've pushed up two patches that make this change in that tree as PR #4147. Could you check they match what you'd expect?

**Comment by rswarbrick** [comment]
Hi Cindy! Just so you know, I've pushed a change to the private CI which should *hopefully* now allow "smoke" or "sanity" as the thing to run (it uses `dvsim.py -l` to figure out which exists). Next time you push, it might Just Work. Or maybe not, in which case I'll spend some time tomorrow figuring out what went wrong :-)

File: hw/dv/doc/dv_plan_template.md (hunk 1 above)
Comment (cindychip) [reply hw/dv/doc/dv_plan_template.md:112]: Thanks Rupert, I will update that.

**Comment by cindychip** [comment]
Thanks Rupert for updating OTBN and update the private-CI. I think it is a good point to let author (at least) review their own block for sure.
How about each IP owner review their IPs. From there we can:
1. Comment if I missed anything small.
2. Or open separate PRs to update the naming for your entire IP(s). (I will remove the changes I made from this PR)
3. Or if your IPs looks good and do not need any updates, please let me know as well.
Let me know what you think. @udinator @sriyerg @weicaiyang @rasmus-madsen @tunghoang290780 @senelson7

File: hw/ip/aes/doc/checklist.md [hunk 2]
```diff
@@ -41,7 +41,7 @@ Documentation | [DOC_INTERFACE][]       | Not Started |
 Documentation | [MISSING_FUNC][]        | Not Started |
 Documentation | [FEATURE_FROZEN][]      | Not Started |
 RTL           | [FEATURE_COMPLETE][]    | Not Started |
-RTL           | [AREA_SANITY_CHECK][]   | Not Started |
+RTL           | [AREA_CROSS_CHECK][]    | Not Started |
```
Comment (cindychip) [inline hw/ip/aes/doc/checklist.md:44]: Thanks @msfschaffner for proposing this change. @eunchan @tjaychen let me know if you prefer something different.

**Review by imphil — APPROVED** [review]
I did scroll through all the changes and didn't see anything obviously wrong. Thanks Cindy!

**Comment by cindychip** [comment]
Thanks @imphil for taking a look at the massive PR. I will close it for now because we might need a little bit more discussion on the DV side. Thanks again and sorry for the noise.

**Comment by msfschaffner** [comment]
Thanks for doing this @cindychip, LGTM!

**Pull request closed without merging**